help_docs: Update invite users for setting invitation expirations.

laurynmm opened the pull request:
Updates documentation on sending invitations to new users to include adding an expiration time period to the invitation as part of the instructions. Also, clarifies the instructions for adding domains in the enable email signup section.

Finally, simplifies some descriptive / instructional text to follow current help center documentation styles (simplifying text when possible, removing default if possible, using shared save-changes doc, etc.).

Fixes #21520.

**Screenshot of updated Send invite instructions block**:
![Screenshot from 2022-03-24 14-43-27](https://user-images.githubusercontent.com/63245456/159929404-e026eb40-4904-42a9-a484-3039bbca9084.png)

**Screenshot of updated Enable email signup instructions block**:
![Screenshot from 2022-03-24 14-42-24](https://user-images.githubusercontent.com/63245456/159929206-5d601746-afbf-4195-817d-084d0e149cab.png)

alya commented on templates/zerver/help/invite-new-users.md:
```diff
@@ -11,9 +10,8 @@ the article below describes each in more detail.
-[role (owner, admin, moderator, member, or guest)](/help/roles-and-permissions) that the
-invited people will have.
+The last two, invite-based techniques also allow you to control the
+[role](/help/roles-and-permissions) that the invited people will have.
```
Let's make this a little easier to read. Something like:

> When sending an invitation, you can also control the
[role](/help/roles-and-permissions) that the invited users will have when they join.

alya commented:
This makes me realize that the whole "Enable email signups" section is really confusing. We should split out the instructions for restricting the domain from general instructions for setting your org to have email signups. I think tabs are not a good mechanism for communicating this.

Could you take a pass at that bit of restructuring?

timabbott commented:
Yeah I'll second that thought -- the precise tabs structure we had here made it feel really hard for us to link to this feature in our recent blog post about a security issue involving this code path.

alya commented:
Let's remove the explicit list of roles under "send invitations".

alya commented:
Let's move the shared "You will only see Invite users in the gear menu if you have permission to invite users." note out of the tabs. I would put something like this just under the "Send invitations" heading:

> You will only see an **Invite users** option if you have permission to invite users to the organization.

laurynmm commented:
So, I've made the updates to the 3 review comments above related to sending invitations.

As far as restructuring the **Enable email signup** section:
- I've removed the tabbed sections that were in the **Enable email signup** section and created two sections instead.
- I've updated the instruction for choosing between the **Restrict email domains of new users?** so that we can add some explanatory text for those options.
- I've also added a second commit that updates the help center documentation related to inviting users for all the updates we've made to the `invite-new-users` article.

See [this CZO chat](https://chat.zulip.org/#narrow/stream/19-documentation/topic/invite.20new.20users.20documentation/near/1352453) for more details.

@alya - Would you be up for updating this current, in-progress version for the explanatory text that's missing from the instruction for the **Restrict email domains of new users** dropdown options (which is in both `invite-new-users` and `allow-anyone-to-join-without-an-invitation`)?

laurynmm commented:
Replaced by #21566.
